parcels/particle.py

```diff
@@ -88,31 +88,33 @@ def AdvectionRK45(particle, grid, time, dt):
         return KernelOp.FAILURE
 
 
-def positions_from_density_field(pnum, startfield, mode='monte_carlo'):
-    # initialise particles from a field
-    print("Initialising particles from " + startfield.name + " field")
-    total = np.sum(startfield.data[0, :, :])
-    startfield.data[0, :, :] = startfield.data[0, :, :]/total
-    lonwidth = (startfield.lon[1] - startfield.lon[0])/2
-    latwidth = (startfield.lat[1] - startfield.lat[0])/2
-
-    def jitter_pos(pos, width, list=[]):
-        list[-1] = pos + np.random.uniform(-width, width)
-        return list
-
-    if(mode is 'monte_carlo'):
+def positions_from_density_field(pnum, field, mode='monte_carlo'):
+    """Initialise particles from a given density field"""
+    print("Initialising particles from " + field.name + " field")
+    total = np.sum(field.data[0, :, :])
+    field.data[0, :, :] = field.data[0, :, :] / total
+    lonwidth = (field.lon[1] - field.lon[0]) / 2
+    latwidth = (field.lat[1] - field.lat[0]) / 2
+
+    def add_jitter(value, width, min, max):
+        value += np.random.uniform(-lonwidth, lonwidth)
+        while not (min <= value <= max):
+            value += np.random.uniform(-width, width)
+        return value
+
```

Comment:
In the case of `value` being outside min or max, should we not re-jitter `value` from it's original position, rather than add yet another random jitter? `value` could conceivably move further and further away from our defined min or max this way.